tools/ios-eddystone-scanner-sample/EddystoneScannerSample/ESSEddystone.m

```diff
@@ -167,23 +168,25 @@ + (instancetype)beaconInfoForUIDFrameData:(NSData *)UIDFrameData
 
   ESSEddystoneUIDFrameFields uidFrame;
 
-  if ([UIDFrameData length] != sizeof(ESSEddystoneUIDFrameFields)) {
-    return nil;
-  }
-  [UIDFrameData getBytes:&uidFrame length:sizeof(ESSEddystoneUIDFrameFields)];
+  if ([UIDFrameData length] == sizeof(ESSEddystoneUIDFrameFields) || [UIDFrameData length] == sizeof(ESSEddystoneUIDFrameFields) - 2) {
+ 
+      [UIDFrameData getBytes:&uidFrame length:(sizeof(ESSEddystoneUIDFrameFields) - sizeof(uidFrame.RFU))];
 
-  NSData *beaconIDData = [NSData dataWithBytes:&uidFrame.beaconID
-                                        length:sizeof(uidFrame.beaconID)];
-  ESSBeaconID *beaconID = [[ESSBeaconID alloc] initWithType:kESSBeaconTypeEddystone
-                                                   beaconID:beaconIDData];
-  if (beaconID == nil) {
-    return nil;
-  }
+      NSData *beaconIDData = [NSData dataWithBytes:&uidFrame.beaconID
+                                            length:sizeof(uidFrame.beaconID)];
+      ESSBeaconID *beaconID = [[ESSBeaconID alloc] initWithType:kESSBeaconTypeEddystone
+                                                       beaconID:beaconIDData];
+      if (beaconID == nil) {
+        return nil;
+      }
 
-  return [[ESSBeaconInfo alloc] initWithBeaconID:beaconID
-                                         txPower:@(uidFrame.txPower)
-                                            RSSI:RSSI
-                                       telemetry:telemetry];
+      return [[ESSBeaconInfo alloc] initWithBeaconID:beaconID
+                                             txPower:@(uidFrame.txPower)
+                                                RSSI:RSSI
+                                           telemetry:telemetry];
+  } else {
```

Comment:
Excluding the else throws the exception `Control may reach end of non-void function`. Are you suggesting  to remove the else and just return nil outside of the conditional statement?

@marcwan Are these the changes you were referring to?